### PR TITLE
publish using github release version

### PR DESCRIPTION
**Pull request by gita-codaio**
## Description:
Previously workflow did not run on main because the key `setup: true` does not allow multiple workflows.
Tested by creating tag `v2.6.15` and setting it to `gv-main-branch-filter` -- see pipeline [here](https://app.circleci.com/pipelines/github/coda/circleci-utils/511/workflows/7965de26-bf65-4c24-8dfd-affa89ce136d) that triggered a publish workflow
[From Circle docs:
](https://support.circleci.com/hc/en-us/articles/360060934851-Troubleshooting-Error-Message-Max-number-of-workflows-exceeded-)
New version created [here](https://circleci.com/developer/orbs/orb/coda/utils?version=2.6.15)
```
CircleCI Dynamic Configuration, formally known as Setup Workflow, allows for a maximum of 1 workflow in the initial setup workflow. So if setup: true is set then the following config would be invalid:
Note: More than one workflow can be defined within a dynamic config, and the above error can be avoided as long as conditional parameters are set so that only one workflow will be executed.
```
To fix we added a filter mechanism on the tag level; this release tag also corresponds to the new version of the orb.

Also updated the docs and added in a filter mechanism to work with the git tag; since the git tags are not able to be filter it just looks if it begins with `v`; however, the tag needs to have the format of `vX.X.X` in order to publish a new version. See[ CI thread here](https://discuss.circleci.com/t/cant-trigger-workflow-on-git-tag-push-using-when-condition/43252/6)
We have to add this filter on both the `workflow` level and the `job` level of the `publish` workflow because there is a bug in [CircleCI] (https://github.com/circleci/circleci-docs/issues/6148#issuecomment-1053874240)

**The New Workflow for Publishing Production Version Orbs**
When an orb is merged the author needs to create a new release tag off of main. After it is created, this new tag will trigger a pipeline run to  publish the new orb with the same version as the github tag version. The version must be in the format of `vX.X.X`
Previously; we would update the version of the orb using the PR body and including a semver version in the title; the orb would read the PR's title and update based on that. 
To Do:
add descrion for /continue workflow and test-deploy
Note: 
Lots of commits because each git tag needed a new commit to test

The `continuation` in the `lint-pack` workflow is what triggers the `test-deploy` yaml file. The workflow continues by trigger everything in the `lint-pack` and when it is completed will automatically trigger `test-deploy`.  Full description of how this workflow work is described in `orb-tools` [here](https://circleci.com/developer/orbs/orb/circleci/orb-tools#usage-step1_lint-pack)

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -64,16 +67,22 @@ workflows:
                   ignore: main
 
   publish:
+    when: << pipeline.git.tag >>
```
Comment (gita-codaio) [inline .circleci/config.yml:70]: this workflow will only run when a new tag is created; this tag corresponds to the version of the orb

File: .circleci/config.yml [hunk 2]
```diff
@@ -35,6 +35,9 @@ parameters:
 
 workflows:
   lint-pack:
+    when:
+      not:
+        equal: [ main, << pipeline.git.branch >> ]
```
Comment (harisiva-codaio) [inline .circleci/config.yml:40]: Doesn't this mean that this job will never run?

**Comment by harisiva-codaio** [comment]
Is the recommended pattern to create the tag to have the author do it locally, and then push it up to `github`?

File: .circleci/config.yml [hunk 3]
```diff
@@ -48,7 +51,7 @@ workflows:
           requires:
             - orb-tools/pack
             - shellcheck/check
-          vcs-type: github
+          vcs-type: << pipeline.project.type >>
```
Comment (harisiva-codaio) [inline .circleci/config.yml:54]: Why this change?

File: .circleci/test-deploy.yml [hunk 4]
```diff
@@ -50,6 +50,9 @@ jobs:
 
 workflows:
   test-deploy:
+    when:
```
Comment (harisiva-codaio) [inline .circleci/test-deploy.yml:53]: How does this `test-deploy` pipeline get used?

File: .circleci/config.yml (hunk 2 above)
Comment (gita-codaio) [reply .circleci/config.yml:40]: Itll only be ran on personal branches, so this is the check before you can merge

**Comment by gita-codaio** [comment]
> Is the recommended pattern to create the tag to have the author do it locally, and then push it up to `github`?

I think it's easier to create tags off of the UI personally, that's how I instructed it in the README; but they could create it locally

File: .circleci/test-deploy.yml (hunk 4 above)
Comment (gita-codaio) [reply .circleci/test-deploy.yml:53]: after `lint-pack` is done, it will trigger `test-deploy`, so it is part of the personal branch pipeline

File: .circleci/config.yml (hunk 3 above)
Comment (gita-codaio) [reply .circleci/config.yml:54]: `pipeline.project.type` is always defined in Circle, I think it's cleaner to use envvar than have it hardcoded

**Comment by gita-codaio** [comment]
> > Is the recommended pattern to create the tag to have the author do it locally, and then push it up to `github`?
> 
> I think it's easier to create tags off of the UI personally, that's how I instructed it in the README; but they could create it locally

but the overall pattern to have it be updated this way was from [`orb-tools` ](https://circleci.com/developer/orbs/orb/circleci/orb-tools#jobs-publish)

File: .circleci/config.yml [hunk 5]
```diff
@@ -48,7 +51,7 @@ workflows:
...
       - orb-tools/continue:
```
Comment (harisiva-codaio) [inline .circleci/config.yml:55]: Can you add a comment here referencing the test-deploy.yml file and the workflow there?

File: .circleci/test-deploy.yml (hunk 4 above)
Comment (harisiva-codaio) [reply .circleci/test-deploy.yml:53]: Can you describe the handshake between this and the main `config.yml` file, and that this is necessitated by circle.

**Review by harisiva-codaio — APPROVED** [review]
